clai/clai/chat/agent_discovery.py

```diff
@@ -0,0 +1,186 @@
+"""Agent discovery using AST parsing to find pydantic_ai.Agent objects."""
```

Comment:
This is cool but I think it's too much magic, at least for the first version of this feature (I wouldn't throw the code away; we could consider it as a separate PR later). I think the existing `-a AGENT` flag is sufficient for now.